pyintegration/deephaven2/calendars.py

```diff
@@ -0,0 +1,372 @@
+    FRIDAY = _JDayOfWeek.FRIDAY
+    SATURDAY = _JDayOfWeek.SATURDAY
+    SUNDAY = _JDayOfWeek.SUNDAY
+
+
+class BusinessPeriod(JObjectWrapper):
+    """ A period of business time during a business day. """
+
+    j_object_type = _JBusinessPeriod
+
+    def __init__(self, j_business_period):
+        self.j_business_period = j_business_period
+
+    def __repr__(self):
+        return f"[{self.start_time}, {self.end_time}]"
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_business_period
+
+    @property
+    def start_time(self) -> DateTime:
+        """ The start of the period. """
+        return self.j_business_period.getStartTime()
+
+    @property
+    def end_time(self) -> DateTime:
+        """ The end of the period. """
+        return self.j_business_period.getEndTime()
+
+    @property
+    def length(self) -> int:
+        """ The length of the period in nanoseconds. """
+        return self.j_business_period.getLength()
+
+
+class BusinessSchedule(JObjectWrapper):
+    """ A business schedule defines a single business day. """
+    j_object_type = _JBusinessSchedule
+
+    def __init__(self, j_business_schedule):
+        self.j_business_schedule = j_business_schedule
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_business_schedule
+
+    @property
+    def business_periods(self) -> List[BusinessPeriod]:
+        """ The business periods of the day.
+
+        Returns:
+             List[BusinessPeriod]
+        """
+        j_periods = self.j_business_schedule.getBusinessPeriods()
+        periods = []
+        for j_period in j_periods:
+            periods.append(BusinessPeriod(j_period))
+        return periods
+
+    @property
+    def start_of_day(self) -> DateTime:
+        """ The start of the business day. """
+        return self.j_business_schedule.getStartOfBusinessDay()
+
+    @property
+    def end_of_day(self) -> DateTime:
+        """ The end of the business day. """
+        return self.j_business_schedule.getEndOfBusinessDay()
+
+    def is_business_day(self) -> bool:
+        """ Whether it is a business day.
+
+        Returns:
+            bool
+        """
+        return self.j_business_schedule.isBusinessDay()
+
+    def is_business_time(self, time: DateTime) -> bool:
+        """ Whether the specified time is a business time for the day.
+
+        Args:
+             time (DateTime): the time during the day
+
+        Return:
+            bool
+        """
+        return self.j_business_schedule.isBusinessTime(time)
+
+    def business_time_elapsed(self, time: DateTime) -> int:
+        """ Returns the amount of business time in nanoseconds that has elapsed on the given day by the specified
+        time.
+
+        Args:
+            time (DateTime): the time during the day
+
+        Returns:
+            int
+        """
+        return self.j_business_schedule.businessTimeElapsed(time)
+
+
+class BusinessCalendar(JObjectWrapper):
```

Comment:
As things are implemented, we have lost a hunk of important stuff.  In the calendar design, there are `Calendars`, and there are `BusinessCalendars`.  `Calendar`s are basic calendars.  `BusinessCalendar`s extend `Calendar`s with the concept of when people are and are not working.  The implementation here just implements the `BusinessCalendar` stuff and not the `Calendar` stuff.  As a result, key functionality, such as `previousDay` is lost.

https://github.com/deephaven/deephaven-core/blob/5c1b52c85ab66a3fb77a0127095453276d024fe2/engine/time/src/main/java/io/deephaven/time/calendar/Calendar.java#L32
https://github.com/deephaven/deephaven-core/blob/5c1b52c85ab66a3fb77a0127095453276d024fe2/engine/time/src/main/java/io/deephaven/time/calendar/BusinessCalendar.java